src/test/groovy/org/osiam/test/integration/ChangeEmailIT.groovy

```diff
@@ -100,6 +106,19 @@ class ChangeEmailIT extends AbstractIT {
         Extension extension = user.getExtension(urn)
         extension.getField('emailConfirmToken', ExtensionFieldType.STRING) != null
         extension.getField('tempMail', ExtensionFieldType.STRING) == newEmailValue
+
+        //Waiting at least 5 seconds to ensure that an email was send
+        Thread.sleep(5000)
```

Comment:
you could actively poll for new emails. No need for a latch, if you don't poll in a background thread.